vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/TreeGrid.java

```diff
@@ -139,6 +140,7 @@ public void commit(int updateId, String parentKey, int levelSize) {
         // Approximated size of the viewport. Used for eager fetching.
         private final int EAGER_FETCH_VIEWPORT_SIZE_ESTIMATE = 40;
         private int viewportRemaining = 0;
+        private List<JsonValue> queuedParents = new ArrayList<>();
```

Comment:
```suggestion
        private final List<JsonValue> queuedParents = new ArrayList<>();
```
Since this is cleared instead of setting a new instance, it can be final. `updateQueueFactory` above could also be made final.